File: src/main/java/com/cesello/command/CommandDemo.java
package com.cesello.command;

import java.util.ArrayList;
import java.util.List;

// client
public class CommandDemo {

    public static void main(String[] args) {
        Light bedroomLight = new Light();
        Light kitchenLight = new Light();
        Switch lightSwitch = new Switch();

//        Command onCommand = new OnCommand(bedroomLight);
//        lightSwitch.storeAndExecute(onCommand);

        Command toggleCommand = new ToggleCommand(bedroomLight);
//
        lightSwitch.storeAndExecute(toggleCommand);
//        lightSwitch.storeAndExecute(toggleCommand);
//        lightSwitch.storeAndExecute(toggleCommand);

        List<Light> lights = new ArrayList<>();
        lights.add(bedroomLight);
        lights.add(kitchenLight);

        Command allLightsCommand = new AllLightsCommand(lights);

        lightSwitch.storeAndExecute(allLightsCommand);
//        lightSwitch.storeAndExecute(allLightsCommand);
//        lightSwitch.storeAndExecute(allLightsCommand);
    }
}

File: src/main/kotlin/com/cesello/command/CommandDemo.kt
package com.cesello.command

fun main() {
    val bedroomLight = Light()
    val kitchenLight = Light()
    val lightSwitch = Switch()

//    val onCommand = OnCommand(bedroomLight)
//    lightSwitch.storeAndExecute(onCommand)

    val toggleCommand = ToggleCommand(bedroomLight)
    lightSwitch.storeAndExecute(toggleCommand)
//    lightSwitch.storeAndExecute(toggleCommand)
//    lightSwitch.storeAndExecute(toggleCommand)

    val lights = listOf(bedroomLight, kitchenLight)

    val allLightsCommand = AllLightsCommand(lights)
    lightSwitch.storeAndExecute(allLightsCommand)
}